src/components/screens/vacation-requests-screen.tsx

```diff
@@ -233,6 +247,16 @@ const VacationRequestsScreen = () => {
     if (!loggedInUser) return;
     try {
       setLoading(true);
+      const unspentDays = Number(
+        getDays(loggedInUser.attributes?.unspentVacationDaysByYear, currentYear)
+      );
+      const requestedDays = Number(vacationRequestData.days ?? 0);
+      const { valid, errorMessage } = validateVacationRequestDays(requestedDays, unspentDays);
+      if (!valid) {
+        setError(errorMessage);
+        setLoading(false);
+        return;
+      }
```

Comment:
Why not extract this into a named reusable function/ utility as it's used multiple times.